src/Network/Network/LoadBalancer/NewAzureLoadBalancerBackendPool.cs

```diff
@@ -0,0 +1,167 @@
+            }
+
+            BackendAddressPool existingloadBalancerBackendAddressPool = null;
+
+            // Confirm if resource already exists
+            try
+            {
+                existingloadBalancerBackendAddressPool = this.NetworkClient.NetworkManagementClient.LoadBalancerBackendAddressPools.Get(this.ResourceGroupName, this.LoadBalancerName, this.Name);
+            }
+            catch (Rest.Azure.CloudException exception)
+            {
+                if (exception.Response.StatusCode == System.Net.HttpStatusCode.NotFound)
+                {
+                    existingloadBalancerBackendAddressPool = null;
+                }
+                else
+                {
+                    throw;
+                }
+            }
+
+            // Throw if object already exists
+            if (existingloadBalancerBackendAddressPool != null)
+            {
+                throw new Exception(string.Format("A BackendPool with name '{0}' in resource group '{1}' already exists under LoadBalancer '{2}'. Please use Set-AzLoadBalancerBackendAddressPool to update an existing BackendPool .",
+                    this.Name, this.ResourceGroupName, this.LoadBalancerName));
+            }
+
+            // Prompt a what if action to user, user can skip with force parameter
+            ConfirmAction(
+                Force.IsPresent,
+                string.Format(Properties.Resources.CreatingResourceMessage, Name),
+                Properties.Resources.CreatingResourceMessage,
+                this.Name,
+                () =>
+                {
+                    var backendPool = CreatePsBackendPool();
+                    WriteObject(backendPool);
+                });
+        }
+
+        private PSBackendAddressPool CreatePsBackendPool()
+        {
+            var backendAddressPool = new BackendAddressPool();
+
+            //Include backend address pool IP's if provided by user
+            if (this.IsParameterBound(c => c.LoadBalancerBackendAddress))
+            {
+                backendAddressPool.LoadBalancerBackendAddresses = new List<LoadBalancerBackendAddress>();
+
+                foreach (var psBackendAddress in LoadBalancerBackendAddress)
+                {
+                    var backendAddress = ToLoadBalancerBackendAddress(psBackendAddress);
+                    backendAddressPool.LoadBalancerBackendAddresses.Add(backendAddress);
+                }
+            }
+
+            var loadBalancerBackendAddressPool = this.NetworkClient.NetworkManagementClient.LoadBalancerBackendAddressPools.CreateOrUpdate(this.ResourceGroupName, this.LoadBalancerName, this.Name, backendAddressPool);
+            var loadBalancerBackendAddressPoolModel = NetworkResourceManagerProfile.Mapper.Map<PSBackendAddressPool>(loadBalancerBackendAddressPool);
+            loadBalancerBackendAddressPool.LoadBalancerBackendAddresses = loadBalancerBackendAddressPool.LoadBalancerBackendAddresses.ToList();
```

Comment:
why do we need to map some children explicitly?

Reply:
Originally I was facing a issue when we added NetworkInterfaceIpConfiguration, but its not needed now and I updated it to use the map.